### DISCUSSION TITLE
Guide for best practices when prompting?

### DISCUSSION THREAD

**Discussion by Kh4rj1** [lllyasviel/Fooocus#1845] [General]
Is there an 'official' guide for writing effective prompts to get the desired results? I've found several online resources, but I'm unsure how applicable they are to Fooocus, particularly because of the prompt expansion.

For example I'd like to know how to get Fooocus to generate two distinct characters with different specified characteristics without it just merging the attributes for both figures.

Thanks!

**Comment by Kh4rj1** [comment]
I'm thinking of correct syntax as well, to what extent spacing, commas, linebreaks and parentheses actually matter

**Comment by xhoxye** [comment]
It is difficult to generate two different characters with different characteristics in a single frame

**Comment by poor7** [comment]
Need additional symbols are required to combine/split prompts. Everything is grouped together without distinction.

**Comment by Amit30swgoh** [comment]
Good question.
https://promptomania.com/stable-diffusion-prompt-builder/
